scripts/rpc-regression.sh

```
@@ -0,0 +1,675 @@
+    if command -v bc &> /dev/null; then
+        if (( $(echo "$val1 == 0" | bc -l 2>/dev/null || echo 0) )) && (( $(echo "$val2 == 0" | bc -l 2>/dev/null || echo 0) )); then
+            return 0
+        fi
+
+        # Avoid division by zero
+        if (( $(echo "$val1 == 0 || $val2 == 0" | bc -l 2>/dev/null || echo 1) )); then
+            return 1
+        fi
+
+        # Use bc for floating point comparison
+        local diff=$(echo "scale=20; if ($val1 > $val2) $val1 - $val2 else $val2 - $val1" | bc -l 2>/dev/null || echo "999")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)